edx/edx-app-ios
# OSPR-2472 Add flag OPEN_LINKS_INSIDE_APP_ENABLED

### Description

[OSPR-2472](https://openedx.atlassian.net/browse/OSPR-2472)

This PR adds the flag `OPEN_LINKS_INSIDE_APP_ENABLED` to enable opening links inside the app if the link has the same domain as `API_HOST_URL`. 

A case for why this is useful: an HTML block that is very large and needs to scroll. You add a link at the bottom so you can go to the top of the HTML. In Studio you will define it with `href="#top"` or something similar. With the current implementation, this link is open outside the app and displays an error because you don't have a logged session. With this flag, you avoid this kind of situations.

### How to test this PR

Run the app with the flag `OPEN_LINKS_INSIDE_APP_ENABLED` set as `true`. In Studio set and HTML block with an URL that has the same domain as `API_HOST_URL` and test that the link open insides the app. If the URL does not belong to the same domain the current behavior should be the same. 

### Reviewers

cc @marcotuts @staubina
